client/src/hoc/PinnedContent.js

```diff
@@ -13,123 +14,130 @@ import text from "./PinnedContent.yaml";
 
 const text_maker = create_text_maker(text);
 
-export class PinnedContent extends React.Component {
-  constructor(props) {
-    super(props);
+export const PinnedContent = withRouter(
+  class extends React.Component {
+    constructor(props) {
+      super(props);
 
-    const local_storage_name = props.local_storage_name;
+      const {
+        match: {
+          params: { options: panel_link },
+        },
+      } = props;
 
-    let user_enabled_pinning;
-    if (has_local_storage) {
-      try {
-        user_enabled_pinning = JSON.parse(
-          localStorage.getItem(local_storage_name)
-        );
-        user_enabled_pinning = _.isBoolean(user_enabled_pinning)
-          ? user_enabled_pinning
-          : true;
-      } catch {
-        user_enabled_pinning = true;
+      const local_storage_name = props.local_storage_name;
+
+      let user_enabled_pinning;
+      if (has_local_storage) {
+        try {
+          user_enabled_pinning = JSON.parse(
+            localStorage.getItem(local_storage_name)
+          );
+          user_enabled_pinning = _.isBoolean(user_enabled_pinning)
+            ? user_enabled_pinning
+            : true;
+        } catch {
+          user_enabled_pinning = true;
+        }
       }
-    }
 
-    this.state = {
-      user_enabled_pinning: user_enabled_pinning,
-    };
-  }
+      if (panel_link) {
+        user_enabled_pinning = false;
+      }
 
-  componentDidUpdate(prev_props, prev_state) {
-    localStorage.setItem(
-      prev_props.local_storage_name,
-      !prev_state.user_enabled_pinning
-    );
-  }
+      this.state = {
+        user_enabled_pinning: user_enabled_pinning,
+      };
+    }
 
-  pin_pressed = () => {
-    const { user_enabled_pinning } = this.state;
-    this.setState({
-      user_enabled_pinning: !user_enabled_pinning,
-    });
-  };
+    pin_pressed = () => {
+      const { local_storage_name } = this.props;
+      const { user_enabled_pinning } = this.state;
+      this.setState({
+        user_enabled_pinning: !user_enabled_pinning,
+      });
+      localStorage.setItem(local_storage_name, !user_enabled_pinning);
```

Comment:
Can we set the local storage before setState? I don't think it should actually be a problem but just in case it re-renders before local storage